db/database.go

```diff
@@ -100,3 +109,65 @@ func (db *Database) ClearAllOffers() error {
 
 	return nil
 }
+
+// PutSwap puts the given swap in the database.
+func (db *Database) PutSwap(s *swap.Info) error {
+	val, err := json.Marshal(s)
+	if err != nil {
+		return err
+	}
+
+	key := s.ID
```

Comment:
I always assumed these ID values indexing swaps were different from offer IDs, but it looks like they are the same. We could avoid a lot of confusion if we called this `ID` field `OfferID` as well.

So we are mapping the ID value in binary (verses hex) and it maps to a JSON (text based) object. We may need some kind of versioning in the `swap.Info` struct at some point.

Reply:
> I always assumed these ID values indexing swaps were different from offer IDs, but it looks like they are the same. We could avoid a lot of confusion if we called this ID field OfferID as well.
yeah the offer ID just becomes the swap ID. sure, we can rename one to the other? or just call them all `ID`?

> So we are mapping the ID value in binary (verses hex)
what do you mean? the ID types for both types.Offer and swap.Info are types.Hash

Reply:
I wrote that when I was first reviewing the PR and didn't know how the code worked yet. One might expect that since the values are encoded in human readable text (JSON), that the ID's might also be text (hex). At the time, I was just clarifying that it was a binary->text mapping. (Not saying that we need to change it.)